iset.mm

```diff
@@ -150553,6 +150553,277 @@ the additive structure must be abelian (see ~ ringcom ), care must be
+      MNZNZCONZLEDUQLEUNABCUQCUANZUOEDACUBNPUNFQUNUQRUNURRUNUORBCUDNPUNGQULUMUC
+      ULUMUEULDURJUMCURDURSHUFTUGUNUPDEUQULUPDPUMCDUOUOSHUHTUIACUQDEFUQSHUJUK
+      $.
+  $}
+
+  ${
+    dvrass.b $e |- B = ( Base ` R ) $.
+    dvrass.o $e |- U = ( Unit ` R ) $.
+    dvrass.d $e |- ./ = ( /r ` R ) $.
+    dvrass.t $e |- .x. = ( .r ` R ) $.
+    $( An associative law for division.  ( ~ divassap analog.)  (Contributed by
+       Mario Carneiro, 4-Dec-2014.) $)
+    dvrass $p |- ( ( R e. Ring /\ ( X e. B /\ Y e. B /\ Z e. U ) ) ->
+      ( ( X .x. Y ) ./ Z ) = ( X .x. ( Y ./ Z ) ) ) $=
+      ( crg wcel w3a co cfv wceq a1i dvrvald wa cinvr simpr1 simpr2 simpr3 eqid
+      simpl ringinvcl syldan ringass syl13anc cbs cmulr ringcl 3adant3r3 oveq2d
+      cui cdvr 3eqtr4d ) CMNZFANZGANZHENZOZUAZFGDPZHCUBQZQZDPZFGVHDPZDPZVFHBPFG
+      HBPZDPVEUTVAVBVHANZVIVKRUTVDUGZUTVAVBVCUCUTVAVBVCUDZUTVDVCVMUTVAVBVCUEZAC
+      EVGHJVGUFZIUHUIACDFGVHILUJUKVEABCDEVGVFHACULQRVEISZDCUMQRVELSZECUQQRVEJSZ
+      VGVGRVEVQSZBCURQRVEKSZVNUTVAVBVFANVCACDFGILUNUOVPTVEVLVJFDVEABCDEVGGHVRVS
+      VTWAWBVNVOVPTUPUS $.
+
+    $( A cancellation law for division.  ( ~ divcanap1 analog.)  (Contributed
+       by Mario Carneiro, 2-Jul-2014.)  (Revised by Mario Carneiro,
+       2-Dec-2014.) $)
+    dvrcan1 $p |- ( ( R e. Ring /\ X e. B /\ Y e. U ) ->
+      ( ( X ./ Y ) .x. Y ) = X ) $=
+      ( crg wcel w3a co cfv wceq a1i eqid 3adant2 cinvr cmulr simp1 simp2 simp3
+      cbs cui dvrvald oveq1d ringinvcl ringsrg syl unitcld ringass syl13anc cur
+      cdvr csrg unitlinv oveq2d ringridm 3adant3 eqtrd 3eqtrd ) CLMZFAMZGEMZNZF
+      GBOZGDOFGCUAPZPZDOZGDOZFVKGDOZDOZFVHVIVLGDVHABCDEVJFGACUFPQVHHRZDCUBPQVHK
+      RECUGPQVHIRZVJVJQVHVJSZRBCUQPQVHJRVEVFVGUCZVEVFVGUDZVEVFVGUEZUHUIVHVEVFVK
+      AMZGAMVMVOQVSVTVEVGWBVFACEVJGIVRHUJTVHACEGVPVQVHVECURMVSCUKULWAUMACDFVKGH
+      KUNUOVHVOFCUPPZDOZFVHVNWCFDVEVGVNWCQVFCDEWCVJGIVRKWCSZUSTUTVEVFWDFQVGACDW
+      CFHKWEVAVBVCVD $.
+
+    $( A cancellation law for division.  ( ~ divcanap3 analog.)  (Contributed
+       by Mario Carneiro, 2-Jul-2014.)  (Revised by Mario Carneiro,
+       18-Jun-2015.) $)
+    dvrcan3 $p |- ( ( R e. Ring /\ X e. B /\ Y e. U ) ->
+      ( ( X .x. Y ) ./ Y ) = X ) $=
+      ( crg wcel w3a co cur cfv wceq simp1 a1i simp2 cbs cui csrg ringsrg simp3
+      unitcld dvrass syl13anc eqid dvrid 3adant2 oveq2d ringridm 3adant3 3eqtrd
+      syl ) CLMZFAMZGEMZNZFGDOGBOZFGGBOZDOZFCPQZDOZFVAURUSGAMUTVBVDRURUSUTSZURU
+      SUTUAVAACEGACUBQRVAHTECUCQRVAITVAURCUDMVGCUEUQURUSUTUFZUGVHABCDEFGGHIJKUH
+      UIVAVCVEFDURUTVCVERUSBCEVEGIJVEUJZUKULUMURUSVFFRUTACDVEFHKVIUNUOUP $.
+  $}
+
+  ${
+    dvreq1.b $e |- B = ( Base ` R ) $.
+    dvreq1.o $e |- U = ( Unit ` R ) $.
+    dvreq1.d $e |- ./ = ( /r ` R ) $.
+    dvreq1.t $e |- .1. = ( 1r ` R ) $.
+    $( A cancellation law for division.  ( ~ diveqap1 analog.)  (Contributed by
+       Mario Carneiro, 28-Apr-2016.) $)
+    dvreq1 $p |- ( ( R e. Ring /\ X e. B /\ Y e. U ) ->
```

Comment:
Are ~divrid, ~divr1 and ~dvreq1 really "cancellation laws"?

Reply:
Agreed. I have updated the comments, in set.mm and iset.mm, with wording inspired by the text we use for their complex number analogues.